services/wallet/router.go

```diff
@@ -507,10 +509,12 @@ func (r *Router) requireApproval(ctx context.Context, sendType SendType, bridge
 		Data:  data,
 	})
 	if err != nil {
-		return false, nil, 0, nil, err
+		return false, nil, 0, 0, nil, err
 	}
 
-	return true, amountIn, estimate, bridgeAddress, nil
+	l1Fee, _ := r.getL1Fee(ctx, network, token.Address, account, amountIn)
```

Comment:
Require approval is not a eth transaction, here we are calling get L1 fee assuming it is, isn't possible to call getL1Fees witht the data above?

Reply:
@alaibe if we call it on other than the optimism chain we will get an error like L1 fee not supported on this chain and the value will be 0, so we continue the block execution, doesn't need to skip the loop.

Reply:
Thanks for answering and good for me, but this comment is about this specific call which need to buil getL1Fee for require approval not for eth tx